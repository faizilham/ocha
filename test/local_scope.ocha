let a = "out";
let b = "b_out";

print(a);                       // out

if (true) {
    let a = "inner";

    print(a, b);                // inner b_out

    let b = "b_in";

    if (true) {
        let a = "inner2";
        print(a, b);            // inner2 b_in
    }

    let c = "c_in";

    print(a, b, c);             // inner b_in c_in
}

let c = "c_out";

print(a, c);                    // out c_out

fn test(a) {
    print(a);

    if (a) {
        let a = "in_test";
        print(a);

        return true;
    }

    return false;
}

test("start");                  // start \n in_test
