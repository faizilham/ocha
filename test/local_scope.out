out
inner b_out
inner2 b_in
inner b_in c_in
out c_out
start
in_test
